src/mahomaps/overlays/CacheFailedOverlay.java

```diff
@@ -0,0 +1,43 @@
+package mahomaps.overlays;
+
+import java.util.Vector;
+
+import mahomaps.MahoMapsApp;
+import mahomaps.map.Geopoint;
+import mahomaps.ui.Button;
+import mahomaps.ui.ColumnsContainer;
+import mahomaps.ui.FillFlowContainer;
+import mahomaps.ui.IButtonHandler;
+import mahomaps.ui.SimpleText;
+import mahomaps.ui.UIElement;
+
+public class CacheFailedOverlay extends MapOverlay implements IButtonHandler {
+
+	public CacheFailedOverlay() {
+		content = new FillFlowContainer(new UIElement[] { new SimpleText("Caching failed!"),
+				new SimpleText("Please, check that memory is not full."), new ColumnsContainer(new UIElement[] {
+						new Button(MahoMapsApp.text[38], 0, this) }) });
+	}
+
+	public String GetId() {
+		return "cache_fail";
```

Comment:
Overlays with same id will discard each other, this behaviour is used in route-related overlays, for example. Here, we probably don't want to mess it with others.